### ISSUE TITLE
Should we make a Egg Manager?

### ISSUE THREAD

**Issue by vinnymac** [vinnymac/PokeNurse#30]
Not sure if this is possible right now, or if anyone even wants this feature. But I could see it fitting in well with this project. Discussion pertaining to it belongs here.

**Comment by mackhankins** [comment]
I think eggs are surely in the wheel house of evolves and transfers.  Not related to this specifically, but one of my ideas was a timer that showed you how much you had left on a lucky egg.  It's important because once you close the app you have no idea how much time you have left to evolve things.  So I'm constantly back and forth trying to keep my eye on it.  **Pro Tip** in google `countdown 30 minutes` is much easier.

I digress, I guess it's about figuring out what data is available and making an interface for it.  I like it for the future.

**Comment by d-bischoff** [comment]
Does Lucky Egg into Mass Evolce work with the current version if you activate the egg with the official client?

**Comment by vinnymac** [comment]
@d-bischoff I think you are asking if you can activate a lucky egg using the official app are you then able to use PokeNurse to evolve everything. You can definitely evolve Pokemon through PokeNurse, so yes it will work.

**Comment by adamlieb** [comment]
I would definitely love to apply incubators to eggs from the app. +1 for that feature.

If the question is about lucky eggs, I think adding an option to activate a lucky egg (before you mass evolve) would be a good feature as well, but I'd prioritize it lower.

**Comment by d-bischoff** [comment]
Thanks for the reply. It also told me where I was wrong when i posted the question, i thought about lucky eggs but this thread was about hatching eggs, Sorry for posting this here than ^^ Well im just concerned that doing mass evolve + lucky egg i will get banned. This really comes very close to cheating in my opining. Thinking about it though. How big is the deplay between evolutions? (is it randomized for detection purposes?) Want to calculate how many evolutions fit into 30mins with this method.

**Comment by vinnymac** [comment]
@d-bischoff currently the evolves are set to a random number between 25-30 seconds per evolution. Which is similar to the real life time it takes between evolves depending on the performance of your phone, some can even do it faster. Evolving isn't cheating as much as it is convenience.

```
Lucky Egg = 30 mins
Non-New Evolution = 500 xp
Time Per Evolution = 30 seconds or less
```

So

```
((30 minutes) / (30 seconds)) * (500 xp * 2)
60 * 1000 xp = 60,000 xp
```

Which means you will earn at least 60k experience per Lucky Egg. At most if you did one every 20s it would be 90k.

**Comment by d-bischoff** [comment]
Thanks a lot. I've seen people get 70k so is 60k fair enough. Maybe I'll fork and make an adjustable delay tomorrow.

**Comment by d-bischoff** [comment]
Yeah well i found the relevant part of the code. I can easily adjust the range in the code. However my javascript skills are not sufficient to make interface elements that can set this up. Sorry guys, no fork.
For people looking for good values check: https://www.reddit.com/r/TheSilphRoad/comments/4x16ab/total_xp_earned_during_lucky_egg_evolution_spree/

**Comment by iPaulis** [comment]
A lucky egg mode or calculator would be a great feature. I'm definetely looking forward to it.
Related to this, right now it shows how many evolves of each pokemon we can do, how about adding a counter for the total sum of evolves at the bottom? and also a counter of selected pokemons.
